src/java/org/apache/cassandra/index/sasi/SASIIndex.java

```diff
@@ -72,9 +73,7 @@ public SecondaryIndexBuilder getIndexBuildTask(ColumnFamilyStore cfs,
                                                        Set<Index> indexes,
                                                        Collection<SSTableReader> sstablesToRebuild)
         {
-            NavigableMap<SSTableReader, Map<ColumnMetadata, ColumnIndex>> sstables = new TreeMap<>((a, b) -> {
-                return Integer.compare(a.descriptor.generation, b.descriptor.generation);
-            });
+            NavigableMap<SSTableReader, Map<ColumnMetadata, ColumnIndex>> sstables = new TreeMap<>(Comparator.comparing(t -> t.descriptor.generation, SSTableUniqueIdentifierFactory.COMPARATOR));
```

Comment:
We can use `SSTableReader.generationComparator`:
```suggestion
            NavigableMap<SSTableReader, Map<ColumnMetadata, ColumnIndex>> sstables = new TreeMap<>(SSTableReader.generationComparator);
```